sklearn/datasets/tests/test_openml.py

```diff
@@ -0,0 +1,242 @@
+"""Test the openml loader.
+"""
+import json
+import numpy as np
+import os
+import scipy.sparse
+import sklearn
+
+from sklearn.datasets import fetch_openml
+from sklearn.utils.testing import (assert_warns_message,
+                                   assert_raise_message)
+from sklearn.externals.arff import load
+from sklearn.externals.six.moves.urllib.error import HTTPError
+
+
+try:
+    # Python 2
+    from urllib2 import urlopen
+except ImportError:
+    # Python 3+
+    from urllib.request import urlopen
+
+
+def fetch_dataset_from_openml(data_id, data_name, data_version,
+                              expected_observations, expected_features,
+                              expect_sparse):
+    # fetch with version
+    data_by_name_id = fetch_openml(name=data_name, version=data_version,
+                                   cache=False)
+    assert int(data_by_name_id.details['id']) == data_id
+
+    # fetch without version
+    fetch_openml(name=data_name, cache=False)
+    # without specifying the version, there is no guarantee that the data id
+    # will be the same
+
+    # fetch with dataset id
+    data_by_id = fetch_openml(id=data_id, cache=False)
+    assert data_by_id.details['name'] == data_name
+    assert data_by_id.data.shape == (expected_observations, expected_features)
+    assert data_by_id.target.shape == (expected_observations, )
```

Comment:
Should we also check here the expected dtype of the final (full) array?

Reply:
Can do so